components/public-api/typescript/BUILD.yaml

```diff
@@ -4,13 +4,26 @@ packages:
     deps:
       - components/gitpod-protocol:lib
     srcs:
-      - "src/*.ts"
-      - "src/*.js"
+      - src/**
       - package.json
+      - tsconfig.json
     config:
       packaging: library
       dontTest: true
       commands:
         build: ["yarn", "build"]
       yarnLock: ${coreYarnLockBase}/../yarn.lock
       tsconfig: tsconfig.json
+
+  - name: publish
+    type: generic
+    env:
+      - DO_PUBLISH=${publishToNPM}
+    argdeps:
+      - npmPublishTrigger
+    deps:
+      - :lib
+      - components/gitpod-protocol:scripts
+    config:
+      commands:
+        - ["node", "components-gitpod-protocol--scripts/publish.js", "${version}", "components-public-api-typescript--lib/package"]
```

Comment:
How is `node` able to find `publish.js` here (the `components-` don't look like valid paths?)

Reply:
I've no idea! This whole thing looks to be way more complicated than it needs to be but not really well placed to simplify. The code here just mirrors what supervisor-api does